tcpclient/connection.go

```diff
@@ -69,13 +70,51 @@ func (c Connection) String() string {
 
 }
 
+// GetTCPProcessingDuration returns the time spent processing the connection
 func (c Connection) GetTCPProcessingDuration() time.Duration {
-	switch c.status {
-	case ConnectionEstablished:
+	if c.WentOk() {
 		return c.metrics.tcpEstablishedDuration
-	case ConnectionError:
-		return c.metrics.tcpErroredDuration
-	default:
-		return 0
 	}
+	return c.metrics.tcpErroredDuration
+}
+
+func (c Connection) isStatusIn(statuses []ConnectionStatus) bool {
+	for _, s := range statuses {
+		if c.GetConnectionStatus() == s {
+			return true
+		}
+	}
+	return false
+}
+
+// WentOk return true when the Connection is Established or Closed state
+func (c Connection) WentOk() bool {
+	if c.isStatusIn([]ConnectionStatus{ConnectionEstablished, ConnectionClosed}) {
```

Comment:
comment valid for all these functions. Why not returning directly c.isStatusIn([]ConnectionStatus{ConnectionEstablished, ConnectionClosed}) ?

Reply:
👍